org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/UriExtensions.xtend

```diff
@@ -20,21 +22,31 @@ import org.eclipse.emf.common.util.URI
 class UriExtensions {
 
 	def URI toUri(String pathWithScheme) {
-		val javaNetUri = java.net.URI.create(pathWithScheme);
-		val path = javaNetUri.toPath
-		return URI.createURI(path)
+		// URI is used to get path of the uri without scheme
+		var path = URI.createURI(pathWithScheme).path
+		return URI.createURI(path.toPath)
 	}
 
 	def String toPath(URI uri) {
```

Comment:
This method should ensure that the returned value starts with `file:///` if the scheme is `file`. It should work as for URIs created via `UriExtensions.toUri` as well as for URIs created with `URI.createURI` and `URI.createFileURI`. Unfortunately, there are not tests for it.